Week3/homework/step2-6.js

```diff
@@ -2,19 +2,44 @@
 
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
+const arr4d = [[[[1, 2], [3, 4]]], [[[5, 6], [7, 8]]], [[[9, 10], [11, 12]]]];
+const arrSoup = [
+  [[[[1, 2], [[['aa', 'bb']]], [3, [['cc', [[['dd']]]]], 4]]]],
+  [[[5, 6], [[[[[[[[[[[[[[[[['zz']]]]]]]]]]]]]]]]], [7, 8]]],
+  [[[9, 10], [11, 12]]],
+];
+
+// for any nested arrays regardless of dimensions
+const flattenNestedArray = (arr, newArray = []) => {
+  arr.forEach(element => {
+    if (Array.isArray(element)) {
+      flattenNestedArray(element, newArray);
+    } else {
+      newArray.push(element);
+    }
+  });
+  return newArray;
+};
```

Comment:
Some pretty good stuff in here:
- recursive function
- default parameters
- forEach

And as you say, this works for any dimension: 💯